lib/Conch/Controller/User.pm

```diff
@@ -219,6 +216,136 @@ sub change_password ($c) {
 	return 1;
 }
 
+=head2 reset_user_password
+
+Generates a new random password for a user. Global admin only.
+
+Optionally takes a query parameter 'send_password_reset_mail' (defaulting to true), to send an
+email to the user with the new password.
+
+Optionally takes a query parameter 'clear_tokens' (defaulting to true), to also revoke session
+tokens for the user, forcing all their tools to log in again.
+
+=cut
+
+sub reset_user_password ($c) {
+	my $user_param = $c->stash('target_user');
+	my $user =
+		is_uuid($user_param) ? $c->db_user_accounts->lookup_by_id($user_param)
+	  : $user_param =~ /^email\=/ ? $c->db_user_accounts->lookup_by_email($')
+	  : undef;
+
+	# user not found, but secure applications don't let on to the user
+	if (not $user) {
+		$c->log->debug("user lookup for $user_param failed");
+		return $c->status(202);
+	}
+
+	my $new_password = $c->random_string();
+	$c->log->debug('user ' . $c->stash('user')->name . ' resetting password for user ' . $user->name);
```

Comment:
nit: let's bump that up to warn or some such.

Reply:
how about info here - as warn is more "something went differently than expected".

Reply:
fine. Same in User.pm please